model_analyzer/result/result_manager.py

```diff
@@ -275,18 +275,24 @@ def collect_and_sort_results(self, num_models):
                 model.model_name()
                 for model in self._config.model_names[:num_models]
         ]:
-            result_dict = results[model_name]
-            for (model_config, measurements) in result_dict.values():
-                result = ModelResult(model_name=model_name,
-                                     model_config=model_config,
-                                     comparator=comparators[model_name],
-                                     constraints=constraints[model_name])
-                for measurement in measurements.values():
-                    measurement.set_result_comparator(
-                        comparator=comparators[model_name])
-                    result.add_measurement(measurement)
-                self._per_model_sorted_results[model_name].add_result(result)
-                self._across_model_sorted_results.add_result(result)
+            if model_name not in results:
```

Comment:
Why is the `model_name` not profiled until this stage? I think this bug was introduced in the checkpoint PR.